pronotepy/clients.py

```diff
@@ -509,24 +510,65 @@ def homework(
                 out.append(hw)
         return out
 
-    def messages(self) -> List[dataClasses.Message]:
+    def get_recipients(self) -> List[dict]:
+        """
+        Get all possible recipients for new discussion
+
+        Returns
+        -------
+        List[dict]
+        """
+        data = {"onglet": {"N": 0, "G": 3}}
+        recipients = self.post("ListeRessourcesPourCommunication", 131, data)[
+            "donneesSec"
+        ]["donnees"]["listeRessourcesPourCommunication"]["V"]
+        data = {"onglet": {"N": 0, "G": 34}}
+        recipients += self.post("ListeRessourcesPourCommunication", 131, data)[
+            "donneesSec"
+        ]["donnees"]["listeRessourcesPourCommunication"]["V"]
+
+        return [r for r in recipients if r.get("avecDiscussion")]
+
+    def new_discussion(self, subjet: str, message: str, recipient: List) -> None:
+        data = {"objet": subjet, "contenu": message, "listeDestinataires": recipient}
+        self.post("SaisieMessage", 131, data)
+
+    def discussions(self) -> List[dataClasses.Discussion]:
         """
         Gets all the discussions in the discussions tab
 
         Returns
         -------
-        List[Messages]
-            Messages
+        List[Discussion]
         """
-        messages = self.post(
+        discussions = self.post(
             "ListeMessagerie", 131, {"avecMessage": True, "avecLu": True}
         )
         return [
-            dataClasses.Message(self, m)
-            for m in messages["donneesSec"]["donnees"]["listeMessagerie"]["V"]
-            if not m.get("estUneDiscussion")
+            dataClasses.Discussion(self, d)
+            for d in discussions["donneesSec"]["donnees"]["listeMessagerie"]["V"]
+            if d.get("estUneDiscussion")
         ]
 
+    def message(self, id: str) -> dataClasses.Message:
```

Comment:
From what I can see this is only used internally? I'd move it into a private method on `Discussion`. People probably won't have the message id without the whole object anyway.

Reply:
Yes it's true, I hadn't thought of doing it like that.